docs/chaos/latency.md

```diff
@@ -0,0 +1,142 @@
+var optionsDefault = new LatencyStrategyOptions();
+
+// The 60% of calls will be randomly affected.
+var basicOptions = new LatencyStrategyOptions
+{
+    Latency = TimeSpan.FromSeconds(30),
+    Enabled = true,
+    InjectionRate = 0.6
+};
+
+// To use a custom function to generate the latency to inject.
+var optionsWithLatencyGenerator = new LatencyStrategyOptions
+{
+    LatencyGenerator = static args =>
+    {
+        TimeSpan latency = args.Context.OperationKey switch
+        {
+            "DataLayer" => TimeSpan.FromMilliseconds(500),
+            "ApplicationLayer" => TimeSpan.FromSeconds(2),
+            _ => TimeSpan.Zero // when the latency generator returns Zero the strategy won't inject any delay and it will just invoke the user's callback
+        };
+
+        return new ValueTask<TimeSpan>(latency);
+    },
+    Enabled = true,
+    InjectionRate = 0.6
+};
+
+// To get notifications when a delay is injected
+var optionsOnBehaviorInjected = new LatencyStrategyOptions
+{
+    Latency = TimeSpan.FromSeconds(30),
+    Enabled = true,
+    InjectionRate = 0.6,
+    OnLatency = static args =>
+    {
+        Console.WriteLine("OnLatency, Latency: {0}, Operation: {0}.", args.Latency, args.Context.OperationKey);
+        return default;
+    }
+};
+
+// Add a latency strategy with a LatencyStrategyOptions{<TResult>} instance to the pipeline
+new ResiliencePipelineBuilder().AddChaosLatency(optionsDefault);
+new ResiliencePipelineBuilder<HttpStatusCode>().AddChaosLatency(optionsWithLatencyGenerator);
+
+// There are also a handy overload to inject the chaos easily.
+new ResiliencePipelineBuilder().AddChaosLatency(0.6, TimeSpan.FromSeconds(30));
+```
+<!-- endSnippet -->
+
+Example execution:
+
+<!-- snippet: latency-execution -->
+```cs
+var pipeline = new ResiliencePipelineBuilder()
+    .AddChaosLatency(new LatencyStrategyOptions // monkey strategies are usually placed innermost in the pipelines
+    {
+        Latency = TimeSpan.FromSeconds(10),
+        Enabled = true,
+        InjectionRate = 0.6
+    })
+    .AddTimeout(TimeSpan.FromSeconds(5))
+    .AddRetry(new RetryStrategyOptions
+    {
+        ShouldHandle = new PredicateBuilder().Handle<TimeoutRejectedException>(),
+        BackoffType = DelayBackoffType.Exponential,
+        UseJitter = true,  // Adds a random factor to the delay
+        MaxRetryAttempts = 4,
+        Delay = TimeSpan.FromSeconds(3),
+    })
+    .Build();
+```
+<!-- endSnippet -->
+
+## Defaults
+
+| Property              | Default Value | Description                                  |
+| --------------------- | ------------- | -------------------------------------------- |
+| `Latency`             | `30 seconds`  | A timestamp indicating the delay to be injected.      |
+| `LatencyGenerator`    | `null`        | Generates the latency to inject for a given execution.|
+| `OnLatency`           | `null`        | Action executed when latency is injected.             |
```

Comment:
Can you tidy the alignment here too please? Either all the rows should be the same width as each other, or each one should be just long enough for the content in them with a space either side.

Reply:
Done